inc/inventoryruleimport.class.php

```diff
@@ -568,6 +570,11 @@ function findWithGlobalCriteria($input) {
                      OR `[typetable]`.`serial`='".$serial2."')";
                   $_SESSION["plugin_fusioninventory_serialHP"] = $serial2;
 
+               // Support partial match for monitor serial
+               } else if (isset($input['itemtype'])
+                       AND $input['itemtype'] == 'Monitor'
+                       AND $pfConfig->getValue('import_monitor_on_partial_sn') == 1) {
+                  $sql_where_temp = " AND `[typetable]`.`serial` LIKE '%".$input["serial"]."%'";
```

Comment:
I have no idea what a partial SN should look like, but isn't that a little too large?

Reply:
Just added a condition on at least 4 chars on the serial lenght. This should be largely enough for the purpose and will avoid unwanted match on empty or corrupted serials.